src/pages/projectSettings/CreateProjectModal.tsx

```diff
@@ -42,27 +44,50 @@ export const CreateProjectModal: React.VFC<Props> = ({
   });
   const [hasError, setHasError] = useState(true);
 
-  const { data } = useQuery<GetGithubOrgsQuery>(GET_GITHUB_ORGS, {
+  const { data: gitOrgs } = useQuery<GetGithubOrgsQuery>(GET_GITHUB_ORGS, {
     skip: !open,
   });
-  const { spruceConfig: { githubOrgs = [] } = {} } = data ?? {};
+  const { spruceConfig: { githubOrgs = [] } = {} } = gitOrgs ?? {};
 
   const form = modalFormDefinition(githubOrgs);
 
-  const [createProject] = useMutation<
+  const [createProject, { called, data, error, loading }] = useMutation<
     CreateProjectMutation,
     CreateProjectMutationVariables
-  >(CREATE_PROJECT, {
-    onCompleted({ createProject: { identifier } }) {
-      dispatchToast.success(`Successfully created the project: ${identifier}`);
+  >(CREATE_PROJECT, { errorPolicy: "all" });
+
+  useEffect(() => {
+    // onCompleted and onError don't provide sufficient information when used with errorPolicy: 'all', so use hook to manage behavior after confirming modal.
+    // https://github.com/apollographql/apollo-client/issues/6966
+    if (!called || loading) {
+      return;
+    }
+
+    const identifier = data?.createProject?.identifier;
+    if (identifier) {
+      if (error) {
+        warning(
+          `The project was successfully created with the following errors: ${error.message}.`,
+          true,
+          { shouldTimeout: false }
+        );
+      } else {
+        success(`Successfully created the project: ${identifier}`);
+      }
```

Comment:
I'm confused by what you are trying to do here?

Reply:
if there is an error and the identifier is returned, we want to warn because the project was created but won't be able to be enabled 
and it will error or not error normally. 

sophie's pr might explain it better than me https://github.com/evergreen-ci/spruce/pull/1257

Reply:
Oh, interesting thank you for explaining. If this is a pattern we want to adopt maybe we should support it more officially. https://github.com/apollographql/apollo-client/issues/6966#issuecomment-1085801223 cc @sophstad

Reply:
I created EVG-19066 to standardize support for this pattern. I think this PR is fine as-is.

Reply:
SGTM